src/window/file_dialog.c

```diff
@@ -331,6 +345,22 @@ static void button_ok_cancel(int is_ok, int param2)
     strncpy(data.file_data->last_loaded_file, filename, FILE_NAME_MAX - 1);
 }
 
+static void button_sort(int param1, int param2)
+{
+    if (!platform_file_manager_has_stat()) return;
+    if (data.sort_by == 0) {
+        dir_sort_by_modified_time();
+        data.sort_by = 1;
+        return;
+    }
```

Comment:
The `else` should be on the same line as `}`.

Also, you don't need a `return` at the end of the `if` or the `else`, really.